### ISSUE TITLE
While Joining Formspecs

### ISSUE THREAD

**Issue by Lejo1** [minetest/minetest#7462]
##### Issue type
 - Feature request

##### Summary
If its possible to add a registration formspec, why not function to show the player a formspec while/before joining.
This would be very useful for rules and other things which should happen before the real join.
Kicking the player should be possible.
Its needed for judicial thing, so that the client needs to accept something, before he get registered.

**Comment by rubenwardy** [comment]
Same idea, although I gave up on it: #2324

**Comment by Lejo1** [comment]
0.5 adds a Registration formspec, so this is the time to allow such formspecs to mods.

**Comment by ClobberXD** [comment]
Good idea. It'd be nice to have a `register_on_player_registration()` call-back... :+1:

**Comment by ClobberXD** [comment]
Slightly off-topic: The registration formspec text should be exposed as a value in `minetest.conf`, so that it can be set by the administrator/owner.

**Comment by nerzhul** [comment]
it's a nice idea i got 2 years ago, it's a good idea

**Comment by appgurueu** [comment]
> Slightly off-topic: The registration formspec text should be exposed as a value in minetest.conf, so that > it can be set by the administrator/owner.
No should be easily accessible in mods, so forget bout MT.conf.

**Comment by ClobberXD** [comment]
It was stupid of me to suggest making this a conf value. Exposing to mods (maybe via a callback?) is the way to go.
